Parse request x-request-id and expose it in contextual logger

delavet opened the pull request:
Provide an util function to extract `x-request-id` from request when request headers are coming, and expose `x-request-id` in contextual loggers. This mechanism is adde to both EPP and BBR.
This resolves https://github.com/kubernetes-sigs/gateway-api-inference-extension/issues/556.

nirrozenbaum commented:
/ok-to-test

kfswain commented:
Looks great! I assume this has been manually validated?
Just want to verify since we dont have test updates (I get it since its just logging changes, but want to verify)

nirrozenbaum commented on pkg/epp/util/request/headers.go:
```diff
@@ -0,0 +1,22 @@
+	RequestIdHeaderKey = "x-request-id"
+)
+
+func ExtractRequestId(req *extProcPb.ProcessingRequest_RequestHeaders) string {
```
one minor comment - 
I would change the function to `ExtractHeaderValue(req *extProcPb.ProcessingRequest_RequestHeaders, headerKey string) string`. 
this util function may become handy in more places as we make progress (I know there were some discussions about passing headers to scheduler filter/scorer/others).

and then use header key `requtil.RequestIdHeaderKey` when calling the function

kfswain replied:
++

delavet replied:
This suggestion makes a lot of sense! I have updated my code, replacing the function with a more generic `ExtractHeaderValue` function, and added unit tests for it.

delavet commented:
> Looks great! I assume this has been manually validated? Just want to verify since we dont have test updates (I get it since its just logging changes, but want to verify)

Thank you for your suggestions. I have updated my PR by replacing the `ExtractRequestId ` function with a more generic version and added unit tests for this function. As for the other parts of the code, since they only involve logging changes, I just run the existing integration tests and e2e tests to ensure the correctness of the functionality.

ahg-g commented:
/approve
